### PR TITLE
Redis implementation

### PR DESCRIPTION
**Pull request by SylteA**
## Summary

Redis implementation for caching and internal events

## Checklist

- [ ] If endpoints were changed then they have been documented and tested.
    - [ ] I have updated the docmentation to reflect the changes.
    - [ ] I have updated the tests to support the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new endpoint or parameter).
- [ ] This PR is a breaking change (e.g. endpoint or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,17 +72,18 @@ pipenv install --dev
...
 ### Environment variables
 
-Set the environment variables. Start by writing this in a file named `local.env`:
+Start by writing this in a file named `.env`:
 
 ```prolog
-SECRET_KEY=some_random_characters_here
-DB_URI=postgresql://user:password@db:5432/twt
+SECRET_KEY=
+POSTGRES_URI=
```
Comment (takos22) [inline README.md:80]: Add REDIS_URI here?

File: README.md [hunk 2]
```diff
@@ -72,17 +72,18 @@ pipenv install --dev
...
 DISCORD_CLIENT_ID=
 DISCORD_CLIENT_SECRET=
 ```
 
 And fill in the variables with the values below:
 
+- `REDIS_URI` is the Redis server URI.
 - `POSTGRES_URI` is the PostgreSQL database URI.
 - `SECRET_KEY` is the key used for JWT token encoding.
 - `TEST_POSTGRES_URI` is the PostgreSQL database URI for tests.
```
Comment (takos22) [inline README.md:91]: Add TEST_REDIS_URI?

**Review by takos22 — APPROVED** [review]
LGTM, I don't really know how redis works exactly so I hope you implemented correctly, other than that the code seems fine

**Review by Shubhaankar-Sharma — APPROVED** [review]

**Review by mohamed040406 — APPROVED** [review]
:eyes:

**Review by takos22 — APPROVED** [review]

**Pull request merged**